trtools/dumpSTR/filters.py

```diff
@@ -592,25 +697,38 @@ class GangSTRCallBadCI(Reason):
 
     def __init__(self):
         pass
-    def __call__(self, sample):
-        #### If ML estimate outside of CI
-        ml = [int(item) for item in sample["REPCN"]]
-        ci = sample["REPCI"].split(",")
-        for i in range(len(ml)):
-            ml_est = ml[i]
-            ci_low = int(ci[i].split("-")[0])
-            ci_high = int(ci[i].split("-")[1])
-            if ml_est<ci_low or ml_est>ci_high: return ml_est
-        return None
-
-class GangSTRCallRequireSupport(Reason):
+    def __call__(self, record: trh.TRRecord):
+        sample_filter = np.empty((record.GetNumSamples()), dtype=float)
+        sample_filter[:] = np.nan
+        called_samples = record.GetCalledSamples()
+        if not np.any(called_samples):
+            return sample_filter
+        ml = record.format["REPCN"][called_samples]
+        ci = np.char.split(record.format["REPCI"][called_samples], ",")
+        ci = np.stack(ci)
+        ci = np.char.split(ci, '-')
+        ci = np.array(ci.tolist(), dtype=int) # now sample x ploidy x 2 (min, max)
+        filter_per_gt = np.logical_or(ml < ci[:, :, 0], ci[:, :, 1] < ml)
+        filter_indicies = np.any(filter_per_gt, axis=1)
+        if not np.any(filter_indicies):
+            return sample_filter
+        problem_gt_indicies = np.argmax(filter_per_gt[filter_indicies, :],
+                                        axis=1)
+        sample_filter[np.nonzero(called_samples)[0][filter_indicies]] = \
+                ml[filter_indicies, problem_gt_indicies]
+        return sample_filter
+
+'''
+Since this class is experimental, I'm just not converting it from pyvcf to
+cyvcf2 for now
+class GangSTRCallRequireSupport(Reason)
```

Comment:
I think this class (and this flag) is used for Ileena's paper and we probably have to use it again for future projects.

Reply:
I'm trying to get these types of side projects done quicker. I'd prefer to push this now and then get back to this when it's needed.